File: 56.java

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;


public class Solution {
    public int[][] merge(int[][] intervals) {
    // lambda
    Arrays.sort(intervals, (a, b) -> Integer.compare(a[0], b[0]));
    ArrayList<int[]> output = new ArrayList<>();

    int i = 0;
    while (i < intervals.length) {
        int start = intervals[i][0];
        int end = intervals[i][1];

        while (i < intervals.length - 1 && intervals[i + 1][0] <= end) {
            end = Math.max(end, intervals[i + 1][1]);
            i++;
        }
        output.add(new int[]{start, end});
        i++;
    }
    return output.toArray(new int[output.size()][]);
    }

}

File: 56.py
from typing import List


class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        output = []
        length = len(intervals)
        # sorting intervals with the the first number using lambda expression
        intervals.sort(key = lambda x: x[0])
        print(intervals)
        print(output)
        i = 0
        while i < length:
            start = intervals[i][0]
            end = intervals[i][1]
            while i < length - 1 and intervals[i + 1][0] <= end:
                i += 1
                end = max(end, intervals[i][1])
            output.append([start,end])
            print(output)
            i += 1
        return output

s = Solution()
intervals = [[2, 6],[1, 3], [8, 10], [15, 18]]
s.merge(intervals)